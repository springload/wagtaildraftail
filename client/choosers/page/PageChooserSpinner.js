import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  isActive: PropTypes.bool,
  children: PropTypes.node,
};

const defaultProps = {
  isActive: false,
  children: null,
};

const PageChooserSpinner = ({ isActive, children }) =>
  <div className={`loading-mask${isActive ? ' loading' : ''}`}>
    {children}
  </div>;

PageChooserSpinner.propTypes = propTypes;
PageChooserSpinner.defaultProps = defaultProps;

export default PageChooserSpinner;
