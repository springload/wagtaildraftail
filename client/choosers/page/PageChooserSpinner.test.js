import React from 'react';
import { shallow } from 'enzyme';
import PageChooserSpinner from './PageChooserSpinner';

describe('PageChooserSpinner', () => {
  it('renders', () => {
    expect(shallow((
      <PageChooserSpinner>
        Test
      </PageChooserSpinner>
    ))).toMatchSnapshot();
  });

  it('#isActive', () => {
    expect(shallow((
      <PageChooserSpinner isActive={true}>
        Test
      </PageChooserSpinner>
    ))).toMatchSnapshot();
  });
});
