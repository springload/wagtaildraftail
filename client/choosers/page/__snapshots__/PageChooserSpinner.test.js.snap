// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageChooserSpinner #isActive 1`] = `
<div
  className="loading-mask loading"
>
  Test
</div>
`;

exports[`PageChooserSpinner renders 1`] = `
<div
  className="loading-mask"
>
  Test
</div>
`;
